src/api/api.js

```diff
@@ -0,0 +1,52 @@
+import { cardData } from "../assets/data/cardData"
+
+export const deckUrl = 'https://deckofcardsapi.com/api/deck/new/'
+export const drawUrl = (id, remaining) => `https://deckofcardsapi.com/api/deck/${id}/draw/?count=${remaining}`
+
+export const fetchDeck = async () => {
+	const { deck_id, remaining } = await fetchApi(deckUrl)
+	const cards = fetchApi(drawUrl(deck_id, remaining)).then(({ cards }) => {
+		return cleanCards(cards)
+	})
+	return cards
+}
+
+export const fetchApi = async (url) => {
+	return await fetch(url).then((response) => response.json()).catch((error) => console.error('Can not reach server.'))
+}
+
+export const cleanCards = (cards) => {
+	return cards.map((card) => {
+		let { value, suit } = card
+
+		if (value === 'ACE') value = 1
+		if (value === 'JACK') value = 11
+		if (value === 'QUEEN') value = 12
+		if (value === 'KING') value = 13
+
+		if (suit === 'SPADES') suit = 'vitality'
+
+		if (suit === 'DIAMONDS') suit = 'spirit'
+		if (suit === 'CLUBS') suit = 'wisdom'
+		if (suit === 'HEARTS') suit = 'mind'
+
+		const newCard = {
+			code: suit[0] + value,
+			talent: suit,
+			count: parseInt(value)
+		}
+
+		updateCardInfo(newCard)
+		return newCard
+	})
+}
+const updateCardInfo = (card) => {
+	return cardData.map((data) => {
```

Comment:
this is technically a nested iterator. There is a way around this but it's a significant refactor.

Reply:
Are you saying it's nested because I'm calling it from an iteration? 

How would you go about refactoring it?

Reply:
Correct.

This is a super similar problem to the one I wrote about [here](https://alexkio.com/data-cleaning). What you would want to do is first create an object that has all of the cards assigned to unique keys (this *.code looks like what I'd use). so the shape of your data would look something like:
```
cardData = {
  uniqueID1: {
    card1
  },
  uniqueID2: {
    card2
  },
  ...
}
```
then you can use dot and bracket notation to directly access what you need:
```
cards.map(card => {
  return cardData[card.code]
}
```

I can help you with this more another time.

Reply:
I see what you mean! That makes sense. This was code I refactored _**to work*_ but I didn't even notice how wonky it ended up haha. It works which is why I didn't notice it. I will refactor it like you said when I remove the card API